contracts/base/LiquidityManagement.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+import {BalanceDelta} from "@uniswap/v4-core/contracts/types/BalanceDelta.sol";
+import {ILockCallback} from "@uniswap/v4-core/contracts/interfaces/callback/ILockCallback.sol";
+import {TickMath} from "@uniswap/v4-core/contracts/libraries/TickMath.sol";
+import {BalanceDelta} from "@uniswap/v4-core/contracts/types/BalanceDelta.sol";
+import {IPoolManager} from "@uniswap/v4-core/contracts/interfaces/IPoolManager.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/contracts/types/PoolId.sol";
+
+import {LiquidityAmounts} from "../libraries/LiquidityAmounts.sol";
+import {PeripheryImmutableState} from "./PeripheryImmutableState.sol";
+import {PeripheryPayments} from "./PeripheryPayments.sol";
+
+/// @title Liquidity management functions
+/// @notice Internal functions for safely managing liquidity in Uniswap V4
+abstract contract LiquidityManagement is ILockCallback, PeripheryImmutableState, PeripheryPayments {
+    using CurrencyLibrary for Currency;
+    using PoolIdLibrary for PoolKey;
+
+    error PriceSlippage();
+
+    enum CallbackType {AddLiquidity}
+
+    struct CallbackData {
+        CallbackType callbackType;
+        address sender;
+        bytes params;
+    }
+
+    struct AddLiquidityParams {
```

Comment:
It seems weird to have both`AddLiquidityParams` and `IncreaseLiquidityParams`.  Like hard to know why you use one and not the other.

Also, its possible that we actually can just have ModifyLiquidityParams that handle both increase and decrease..

Reply:
And we should be consistent with the naming add/remove vs. increase/decrease. I prefer increase/decrease I think